hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ledger/accounts/AliasManager.java

```diff
@@ -240,7 +244,30 @@ public void forgetEvmAddress(final ByteString alias) {
      * @return EntityNum mapped to the given alias.
      */
     public EntityNum lookupIdBy(final ByteString alias) {
-        return curAliases().getOrDefault(alias, MISSING_NUM);
+        var entityNum = curAliases().getOrDefault(alias, MISSING_NUM);
+        if (alias.size() > EVM_ADDRESS_LEN && entityNum == MISSING_NUM) {
```

Comment:
Is this to protect us from doing an auto-creation with an `secp256k1` key that would collide with an existing EVM address?

Reply:
Added comment.